### PR TITLE
refactor(addon-doc): drop `AsyncPipe`

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

**Comment by splincode** [comment]
@waterplea what do you think?

https://github.com/taiga-family/taiga-ui/pull/9102

File: projects/addon-doc/components/code/index.ts [hunk 1]
```diff
@@ -55,9 +55,13 @@ export class TuiDocCode {
...
     );
 
-    protected readonly processor$ = this.rawLoader$$.pipe(
-        switchMap(tuiRawLoad),
-        map((value: string): string[] => this.markdownCodeProcessor(value)),
+    protected readonly processor = toSignal(
+        this.rawLoader$$.pipe(
+            switchMap(tuiRawLoad),
+            map((value: string): readonly string[] => this.markdownCodeProcessor(value)),
+            startWith([] satisfies readonly string[]),
```
Comment (vladimirpotekhin) [inline projects/addon-doc/components/code/index.ts:62]: is `startWith` necessary?

File: projects/addon-doc/components/example/example-get-tabs.pipe.ts [hunk 2]
```diff
@@ -6,7 +6,10 @@ import {Pipe} from '@angular/core';
     name: 'tuiDocExampleGetTabs',
 })
 export class TuiDocExampleGetTabsPipe implements PipeTransform {
-    public transform(content: Record<string, string>, defaultTab: string): string[] {
+    public transform(
+        content: Record<string, never> | Record<string, string>,
```
Comment (nsbarsukov) [inline projects/addon-doc/components/example/example-get-tabs.pipe.ts:10]: Why do we need to extend this type ?

I've tried to remove it – `nx build demo` still succeeds

File: projects/addon-doc/components/example/example.template.html [hunk 3]
```diff
@@ -96,13 +93,13 @@
...
                 <ng-content />
-                <ng-container *polymorpheusOutlet="lazyComponent$ | async as text">
+                <ng-container *polymorpheusOutlet="lazyComponent() as text">
                     {{ text }}
                 </ng-container>
             </div>
 
             <tui-doc-code
-                *tuiLet="files?.[tabs?.[index] || 0] || '' as code"
+                *tuiLet="processor()[tabs?.[index] || 0] || '' as code"
```
Comment (nsbarsukov) [inline projects/addon-doc/components/example/example.template.html:102]: Do we actually need it ?

```suggestion
                *tuiLet="processor()[tabs[index] || 0] || '' as code"
```

File: projects/addon-doc/components/internal/header/index.ts [hunk 4]
```diff
@@ -33,10 +32,14 @@ export class TuiDocHeader {
...
-        this.router.events.pipe(map(TUI_FALSE_HANDLER)),
-        this.stream$,
-    ).pipe(startWith(false), distinctUntilChanged());
+
+    protected readonly open = toSignal(
+        merge(this.router.events.pipe(map(TUI_FALSE_HANDLER)), this.stream$).pipe(
+            startWith(false),
+            distinctUntilChanged(),
+        ),
+        {initialValue: false},
```
Comment (nsbarsukov) [inline projects/addon-doc/components/internal/header/index.ts:41]: Do we need `startWith(false)` if we have `{initialValue: false}` ?

File: projects/addon-doc/components/copy/index.ts [hunk 5]
```diff
@@ -1,36 +1,36 @@
...
     templateUrl: './index.html',
     styleUrls: ['./index.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TuiDocCopy {
     private readonly copy$ = new Subject<void>();
 
-    protected readonly texts$ = inject(TUI_COPY_TEXTS);
+    protected readonly texts = toSignal(inject(TUI_COPY_TEXTS), {
+        initialValue: ['', ''],
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/components/copy/index.ts:23]: image

File: projects/addon-doc/components/example/example.template.html (hunk 3 above)
Comment (splincode) [reply projects/addon-doc/components/example/example.template.html]: fixed

File: projects/addon-doc/components/example/example-get-tabs.pipe.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-doc/components/example/example-get-tabs.pipe.ts]: fixed

File: projects/addon-doc/components/documentation/documentation-property-connector.directive.ts [hunk 6]
```diff
@@ -92,7 +92,7 @@ export class TuiDocDocumentationPropertyConnector<T> implements OnInit, OnChange
         // For more convenient debugging
         console.info(this.attrName, event);
 
-        this.emits$.next(this.emits$.value + 1);
+        this.emits.set(this.emits() + 1);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/components/documentation/documentation-property-connector.directive.ts]: fixed

File: projects/addon-doc/components/example/example.template.html (hunk 3 above)
Comment (splincode) [reply projects/addon-doc/components/example/example.template.html]: fixed

File: projects/addon-doc/components/code/index.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-doc/components/code/index.ts]: fixed

File: projects/addon-doc/components/copy/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/addon-doc/components/copy/index.ts]: Did you add `as const` ?

**Review by waterplea — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**